Add SimpleCov.

squaresurf opened the pull request:
Why:

* SimpleCov will give us local insight into our test coverage and where
we can improve.

This change addresses the need by:

* Add SimpleCov gem to the test environment.
* Default the COVERAGE env var to on.
* Enable SimpleCov in the spec helper.

Mrjaco12 commented:
We should also add the cov/ dir to the .gitignore, I believe.

samcdavid commented on spec/spec_helper.rb:
```diff
@@ -1,3 +1,11 @@
+  require 'simplecov'
+  SimpleCov.start 'rails' do
+    add_filter '/app/admin/'
+    add_filter '/spec/'
```
I don't believe the spec folder has to be filtered out. I haven't had to filter it out for any of the projects where I have used simplecov.

samcdavid commented:
+1 to @Mrjaco12 comment, the `coverage` directory needs to be added to the gitignore. Other than that it looks great.

squaresurf commented:
@samcdavid I've fixed the spec helper to not exclude the `spec` folder. I had to do that on another project that used more than one test framework, but we shouldn't need that here as we only use rspec currently.

I've also added the coverage directory to the gitignore and cleaned it up a bit.

@samcdavid @Mrjaco12 would you please re-review?

Mrjaco12 commented:
:+1:

samcdavid commented:
:+1:
